nasa/harmony-gdal-adapter
# HGA - Further code clean-up.

## Description

This PR does a couple of things:

* Moves some code from `HarmonyAdapter.process_item` into its own class method `HarmonyAdapter.stage_and_get_output_stac`. This is primarily to increase the readability at the highest level of abstraction for processing an item.
* Slightly streamlines the `HarmonyAdapter.process_zip` and `HarmonyAdapter.unpack_zip` methods to remove unnecessary variables (e.g., you don't need a boolean to indicate if there are files of a certain type, when you can just check if the list of those files is empty).
* Refactors and adds unit tests for `get_files_from_unzipfiles`. This simplifies the logic of the function and use more Python-standard functionality, like list comprehensions. Note - the unit tests were added prior to the refactor to ensure consistent behaviour before and after.

One slight concern - my local Harmony instance is playing up (too many redirects when interacting with EDL). That means this code has not been tested by using the HGA regression test notebook. I'm a little wary of these changes until being able to run those tests again.

## Jira Issue ID

N/A

## Local Test Steps

If your local Harmony instance is still working:
* Pull this branch.
* Build the image locally (`bin/build-image`)
* Restart the services in your local Harmony instance (`bin/restart-services`)
* Run the HGA regression test notebook. All tests should pass. (If the notebook fails when trying to instantiate a harmony-py `Client` object, then you have the same issues as me.

## PR Acceptance Checklist
* ~~Jira ticket acceptance criteria met.~~
* ~~Documentation updated (if needed).~~
